src/btree/row_srch.c

```diff
@@ -102,7 +102,18 @@ __wt_search_insert(
     match = skiphigh = skiplow = 0;
     ins = last_ins = NULL;
     for (i = WT_SKIP_MAXDEPTH - 1, insp = &ins_head->head[i]; i >= 0;) {
-        if ((ins = *insp) == NULL) {
+        /*
```

Comment:
Is it clearer if we say that our intention here is to cache the copy of the insert pointer at the start of the loop iteration? This ensures we do not re-read a conflicting value if it were to change during the iteration. The read barrier also prevents the compiler from optimising the cached value away.

For instance, we could say:

"Do an ordered read to get the insert pointer. A CPU with weak memory ordering, such as ARM and PPC, may re-order insert pointer reads. This might lead to incorrectly using an older value. It is not OK, and the reason is explained in the following comment. We also cache the insert pointer at the beginning of the loop iteration. This ensures we don't re-read a conflicting value if it were to change during the iteration. Using a read barrier also prevents the compiler from optimising away the cached value."